### PR TITLE
Only shlex.split strings

### PR DESCRIPTION
**Pull request by dnd**
The recently added calls to shlex.split can cause improper results when shlex.split tries to split a value that is already a list. This adds a util function `shlex.split` that checks to make sure the value being split is a string.

As an example, a value coming in to `rbenv.do_with_ruby` of `['gem', 'list', 'bundler']`, gets mangled to: `['[[gem,,', 'list,,', 'bundler]]']` by the time it is actually executed upon which time it fails as an invalid command.

### CODE REVIEW

File: salt/utils/__init__.py [hunk 1]
```diff
@@ -2877,3 +2877,12 @@ def invalid_kwargs(invalid_kwargs, raise_exc=True):
         raise SaltInvocationError(msg)
     else:
         return msg
+
+def shlex_split(s, **kwargs):
+    '''
+    Only split if variable is a string
+    '''
+    if isinstance(s, basestring):
```
Comment (cachedout) [inline salt/utils/__init__.py]: Let's make this `salt.ext.six.string_types` instead.

**Comment by cachedout** [comment]
Hi @dnd 

The general idea here looks sound. Great idea!

I'd prefer we use our string types lib here. I've made a suggestion in-line here for you to have a look at. Let me know what you think.

**Comment by dnd** [comment]
I don't know enough to have a problem with that. :) I just saw it when I
was making another chang and just looked like something that could be
cleaned up. I'll update it when I get back from lunch.

**Comment by cachedout** [comment]
@dnd Sounds great. This will be a good addition. Thanks for the work!

**Comment by dnd** [comment]
This all came about from the rbenv module in develop. Now that I'm looking at it again, it's wrapped in a `try` block(https://github.com/saltstack/salt/blob/develop/salt/modules/rbenv.py#L417). Do we know what exactly that was guarding against that the decision was made to just forcibly convert whatever came through there into a string if initial parsing failed?

If there was a reason other than failure on lists, should this be included in the new util function?

**Comment by cachedout** [comment]
I'm afraid I don't know what the original reasoning was in that module. My assumption from reading the code is the same as yours, though.

**Comment by cachedout** [comment]
@dnd There are now unused imports which are generating pylint errors. Could you please have a look and clean them up?

**Comment by dnd** [comment]
@cachedout I cleaned up the lint errors that it showed. I did notice that `salt/modules/dockerng.py` appeared to have an unused `import shlex`, but it didn't appear in the lint errors list so I left it alone.

**Pull request merged**